Source/Core/DolphinQt/Settings/GameCubePane.cpp

```diff
@@ -303,66 +303,86 @@ void GameCubePane::BrowseMemcard(ExpansionInterface::Slot slot)
 
-    if (error_code.HasCriticalErrors() || !mc || !mc->IsValid())
-    {
-      ModalMessageBox::critical(
-          this, tr("Error"),
-          tr("The file\n%1\nis either corrupted or not a GameCube memory card file.\n%2")
-              .arg(filename)
-              .arg(GCMemcardManager::GetErrorMessagesForErrorCode(error_code)));
-      return;
-    }
+  if (!raw_path_valid)
+  {
+    // TODO: We could try to autodetect the card region here and offer automatic renaming.
+    ModalMessageBox::critical(this, tr("Error"),
+                              tr("The filename %1 does not conform to Dolphin's region code format "
+                                 "for memory cards. Please rename this file to either %2, %3, or "
+                                 "%4, matching the region of the save files that are on it.")
+                                  .arg(QString::fromStdString(PathToFileName(raw_path)))
+                                  .arg(QString::fromStdString(PathToFileName(us_path)))
+                                  .arg(QString::fromStdString(PathToFileName(eu_path)))
+                                  .arg(QString::fromStdString(PathToFileName(jp_path))));
+    return;
   }
 
-  for (ExpansionInterface::Slot other_slot : ExpansionInterface::MEMCARD_SLOTS)
+  // Memcard validity checks
+  for (const std::string& path : {jp_path, us_path, eu_path})
   {
-    if (other_slot == slot)
-      continue;
-
-    bool other_slot_memcard = m_slot_combos[other_slot]->currentData().toInt() ==
-                              static_cast<int>(ExpansionInterface::EXIDeviceType::MemoryCard);
-    if (other_slot_memcard)
+    if (File::Exists(path))
     {
-      QString path_other =
-          QFileInfo(QString::fromStdString(Config::Get(Config::GetInfoForMemcardPath(other_slot))))
-              .absoluteFilePath();
+      auto [error_code, mc] = Memcard::GCMemcard::Open(path);
 
-      if (path_abs == path_other)
+      if (error_code.HasCriticalErrors() || !mc || !mc->IsValid())
       {
         ModalMessageBox::critical(
             this, tr("Error"),
-            tr("The same file can't be used in multiple slots; it is already used by %1.")
-                .arg(QString::fromStdString(fmt::to_string(other_slot))));
+            tr("The file\n%1\nis either corrupted or not a GameCube memory card file.\n%2")
+                .arg(QString::fromStdString(path))
+                .arg(GCMemcardManager::GetErrorMessagesForErrorCode(error_code)));
         return;
       }
     }
   }
 
-  QString path_old =
-      QFileInfo(QString::fromStdString(Config::Get(Config::GetInfoForMemcardPath(slot))))
-          .absoluteFilePath();
+  for (ExpansionInterface::Slot other_slot : ExpansionInterface::MEMCARD_SLOTS)
+  {
+    if (other_slot == slot)
+      continue;
+
+    const std::string other_eu_path = Config::GetMemcardPath(other_slot, DiscIO::Region::PAL);
```

Comment:
Does it matter that PAL was used here and the user's card isn't PAL?  I'm guessing that we just picked a region to compare the two slots, is that correct?

Reply:
Yeah, I just picked a random region here to compare, it doesn't actually matter.

Reply:
It's possible I am just dumb but it took me a moment or two to realize that.  It might be nice to add a comment.  Up to you.